### PR TITLE
Make odbc based dialects optional

### LINKED ISSUE #353: 🔧 Make pyodbc an optional dependency (feature) through setuptools extras
# Summary

This is just a thought based on some light interaction with the library, but it would be nice if different dependencies for all the ways of connecting could be made optional somehow.

# Details

## Background & Context

@Nicoretti opened [a nice discussion about adding an ibis backend](https://github.com/ibis-project/ibis/discussions/6448), and it's likely we'll use this library, and ideally we can use the websockets API to avoid ODBC :)

However, it seems that the package unconditionally requires (and therefore installs) pyodbc, even if it never gets used.

Ideally we can avoid giving users a dependency they never use.

## Examples

I think the main changes would be that installation would look different:

```
$ pip install sqlalchemy-exasol[pyodbc]
$ pip install sqlalchemy-exasol[websockets]
$ pip install sqlalchemy-exasol # not sure what this would do, maybe an extra should be required?
```

## Related Issue(s)
* https://github.com/exasol/sqlalchemy-exasol/issues/402

### PR DESCRIPTION
**Pull request by Nicoretti**
fixes #353

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -85,7 +85,7 @@ jobs:
           python-version: ${{ matrix.python }}
 
       - name: Install python project dependencies including trubodbc
```
Comment (tkilias) [inline .github/workflows/ci-cd.yml]:
```suggestion
      - name: Install python project dependencies including turbodbc
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -84,7 +84,7 @@ jobs:
           python-version: ${{ matrix.python }}
 
       - name: Install python project dependencies including trubodbc
```
Comment (tkilias) [inline .github/workflows/ci.yaml]:
```suggestion
      - name: Install python project dependencies including turbodbc
```

File: README.rst [hunk 3]
```diff
@@ -43,68 +43,89 @@ SQLAlchemy Dialect for EXASOL DB
...
+++++++++++++++++++
+
+We recommend using the Websocket-based dialect due to its simplicity. ODBC-based dialects demand a thorough understanding of (Unix)ODBC, and the setup is considerably more complex.
+
+.. note::
 
-Generally, we advise to use the websocket based Dialect, because odbc
-based dialects require a good understanding of (unix)ODBC and the setup is
-significant more complicated.
+   To use an ODBC-based dialect, you must specify it as an extra during installation.
```
Comment (tkilias) [inline README.rst]: should we have a quick installation guide to show how install it without odbc, because this only gets shown way later

**Review by tkilias — CHANGES_REQUESTED** [review]

File: README.rst (hunk 3 above)
Comment (Nicoretti) [reply README.rst]: I think it is fine, because if you do nothing (standard install) you get it.

**Review by tkilias — DISMISSED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**